core/templates/pages/contributor-dashboard-page/contributor-stats/contributor-stats.component.ts

```diff
@@ -0,0 +1,388 @@
+      userContributionRights.can_review_questions);
+    this.userCanSuggestQuestions = (
+      userContributionRights.can_suggest_questions);
+
+    if (this.userCanReviewTranslationSuggestions) {
+      this.options.push(this.translationReviewOption);
+    }
+    if (this.userCanSuggestQuestions) {
+      this.options.push(this.questionContributionOption);
+    }
+    if (this.userCanReviewQuestionSuggestions) {
+      this.options.push(this.questionReviewOption);
+    }
+
+    await this.fetchStats();
+  }
+
+  toggleDropdown(): void {
+    this.dropdownShown = !this.dropdownShown;
+  }
+
+  toggleMobileDropdown(): void {
+    this.mobileDropdownShown = !this.mobileDropdownShown;
+  }
+
+  async selectOption(contributionType: string): Promise<void> {
+    this.type = contributionType;
+    const currentOption = this.options.find(
+      (option) => option.contributionType === contributionType);
+    this.selectedContributionType = currentOption.displayName;
+    this.dropdownShown = false;
+    this.mobileDropdownShown = false;
+  }
+
+  async fetchStats(): Promise<void> {
+    const response = await this.contributionAndReviewStatsService.fetchAllStats(
+      this.username);
+
+    if (response.translation_contribution_stats.length > 0) {
+      response.translation_contribution_stats.map((stat) => {
+        if (!this.statsData.translationContribution[stat.language_code]) {
+          this.statsData.translationContribution[stat.language_code] = this
+            .createTranslationContributionStatsPage(stat);
+        } else {
+          this.statsData.translationContribution[stat.language_code].data.push(
+            this.createTranslationContributionStat(stat));
+        }
+      });
+    }
+
+    if (response.translation_review_stats.length > 0) {
+      response.translation_review_stats.map((stat) => {
+        if (!this.statsData.translationReview[stat.language_code]) {
+          this.statsData.translationReview[stat.language_code] = this
+            .createTranslationReviewStatsPage(stat);
+        } else {
+          this.statsData.translationReview[stat.language_code].data.push(
+            this.createTranslationReviewStat(stat));
+        }
+      });
+    }
+
+    if (response.question_contribution_stats.length > 0) {
+      this.statsData.questionContribution = this
+        .createQuestionContributionStatsPage(
+          response.question_contribution_stats);
+    }
+
+    if (response.question_review_stats.length > 0) {
+      this.statsData.questionReview = this.createQuestionReviewStatsPage(
+        response.question_review_stats);
+    }
+  }
+
+  createTranslationContributionStatsPage(
+      stat: TranslationContributionBackendDict): StatsPage {
+    return {
+      data: [this.createTranslationContributionStat(stat)],
+      language: this.languageUtilService.getAudioLanguageDescription(
+        stat.language_code),
+      currentPageStartIndex: 0,
+      currentPageEndIndex: 5
```

Comment:
```suggestion
      currentPageEndIndex: ITEMS_PER_PAGE 
```

Reply:
Not done? Ditto for all the other `currentPageEndIndex: 5` usages.

Reply:
Removed this filed.